src/Credentials/InstanceProfileProvider.php

```diff
@@ -60,6 +60,9 @@ class InstanceProfileProvider
     /** @var string */
     private $endpointMode;
 
+    /** @var bool */
```

Comment:
This property just feels like it doesn't belong to this class.  Could we set a `$config` property instead and pass that the the resolver?